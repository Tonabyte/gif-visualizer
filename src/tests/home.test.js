import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

// Screens
import Home from '../screens/home';

describe('Home Test Suite', () => {

  it('Should show Home component', () => {
    const wrapper = shallow(<Home />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should show Home component with configured categories', () => {
    const categories = ['sun', 'happiness'];
    const wrapper = shallow(<Home  defaultConfigurations={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Grid').length).toEqual(categories.length);
  });
});
