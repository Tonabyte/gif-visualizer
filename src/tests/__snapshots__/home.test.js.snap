// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Test Suite Should show Home component 1`] = `
<Fragment>
  <h2>
    Gif Visualizer
  </h2>
  <Searchbar
    addCategory={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Home Test Suite Should show Home component with configured categories 1`] = `
<Fragment>
  <h2>
    Gif Visualizer
  </h2>
  <Searchbar
    addCategory={[Function]}
  />
  <hr />
  <ol>
    <Grid
      category="sun"
      key="sun"
    />
    <Grid
      category="happiness"
      key="happiness"
    />
  </ol>
</Fragment>
`;
